scripts/pi-hole/php/teleporter.php

```diff
@@ -125,28 +108,39 @@ function getWildcardListContent() {
 			if(isset($_POST["blacklist"]) && $file->getFilename() === "blacklist.txt")
 			{
 				$blacklist = process_file(file_get_contents($file));
-				echo "Processing blacklist.txt<br>\n";
+				echo "Processing blacklist.txt (".count($blacklist)." entries)<br>\n";
 				exec("sudo pihole -b -nr --nuke");
 				exec("sudo pihole -b -q -nr ".implode(" ", $blacklist));
 				$importedsomething = true;
 			}
 			if(isset($_POST["whitelist"]) && $file->getFilename() === "whitelist.txt")
 			{
 				$whitelist = process_file(file_get_contents($file));
-				echo "Processing whitelist.txt<br>\n";
+				echo "Processing whitelist.txt (".count($whitelist)." entries)<br>\n";
 				exec("sudo pihole -w -nr --nuke");
 				exec("sudo pihole -w -q -nr ".implode(" ", $whitelist));
 				$importedsomething = true;
 			}
 
-			if(isset($_POST["wildlist"]) && $file->getFilename() === "wildcardblocking.txt")
+			if(isset($_POST["regexlist"]) && $file->getFilename() === "regex.list")
+			{
+				$regexlist = process_file(file_get_contents($file),false);
+				echo "Processing regex.list (".count($regexlist)." entries)<br>\n";
+				exec("sudo pihole --regex -nr --nuke");
+				exec("sudo pihole --regex -q -nr ".implode(" ", $regexlist));
```

Comment:
Because we do not validate regex, this could be insecure. The hard thing is that an injected script could be a valid regex. Do you have any ideas on how to secure this? Perhaps we should directly write the regex information to disk instead of going through Bash.